Unreal Engine: Add event fingerprint API docs (#13808)
getsentry/sentry-docs: docs/platforms/unreal/usage/sdk-fingerprinting/index.mdx

---
title: "SDK Fingerprinting"
description: "Learn about overriding default fingerprinting in your SDK."
sidebar_order: 30
---

All events have a fingerprint. Events with the same fingerprint are grouped together into an issue.

By default, Sentry will run one of our built-in grouping algorithms to generate a fingerprint based on information available within the event such as `stacktrace`, `exception`, and `message`. To extend the default grouping behavior or change it completely, you can use a combination of the following options:

1. In your SDK, using SDK Fingerprinting, as documented below
2. In your project, using [Fingerprint Rules](/concepts/data-management/event-grouping/fingerprint-rules/) or [Stack Trace Rules](/concepts/data-management/event-grouping/stack-trace-rules/)

> In supported SDKs, you can override Sentry's default grouping that passes the fingerprint attribute as an array of strings. The length of a fingerprint's array is not restricted. This works similarly to the [fingerprint rules functionality](/concepts/data-management/event-grouping/fingerprint-rules/), which is always available and can achieve similar results.

Reviewer: Is there a specific part of that fingerprint doc you want to be pointing someone to? You link to the same doc from the top down just above, so wanted to double check.

Author: No worries - it's all good. Just wanted to provide the full context here, but appreciate you double-checking!

## Basic Example

In the most basic case, values are passed directly:

<PlatformContent includePath="set-fingerprint/basic" />

> You can use variable substitution to fill dynamic values into the fingerprint generally computed on the server. For instance, the value `{{ default }}` can be added to add the entire normally generated grouping hash into the fingerprint. These values are the same as for server-side fingerprinting. See [Variables](/concepts/data-management/event-grouping/fingerprint-rules/#variables) for more information.

Reviewer: ```suggestion
You can use variable substitution to fill dynamic values into the fingerprint that by default is computed on the server. For example, the value `{{ default }}` can be added to include the entire normally generated grouping hash into the fingerprint. These values are the same for server-side fingerprinting. See [Variables](/concepts/data-management/event-grouping/fingerprint-rules/#variables) for more information.
```

## Group Errors With Greater Granularity

In some scenarios, you'll want to group errors more granularly.

For example, if your application queries a Remote Procedure Call Model (RPC) interface or external Application Programming Interface (API) service, the stack trace is generally the same, even if the outgoing request is very different.

> The following example will split up the default group Sentry would create (represented by `{{ default }}`) further, taking some attributes on the error object into account:

Reviewer: ```suggestion
The following example will split up the default group Sentry would create (represented by `{{ default }}`) even more, taking some attributes on the error object into account:
```
